src/components/elements/input/TextInput.tsx

```diff
@@ -49,7 +49,7 @@ const TextInput = ({
   const htmlId = id || generatedId;
 
   let width = inputWidth;
-  if (!width && inputProps.inputMode === 'numeric') {
+  if (!width && inputProps.inputMode === 'numeric' && !sx) {
```

Comment:
skip this hard-coded width if we are supplying sx, this prevents the width from changing on the input when it's focused in the mui pickers.